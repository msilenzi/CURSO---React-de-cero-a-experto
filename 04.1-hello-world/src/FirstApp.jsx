import PropTypes from 'prop-types'

function FirstApp({ title, subtitle, name }) {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <h2>{subtitle}</h2>
      <p>{name}</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  name: PropTypes.string,
}

FirstApp.defaultProps = {
  subtitle: 'No hay subt√≠tulo',
  name: 'Manuel Silenzi',
}

export default FirstApp
