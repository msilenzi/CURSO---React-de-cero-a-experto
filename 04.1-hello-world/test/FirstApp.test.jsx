import FirstApp from '../src/FirstApp'
import { render } from '@testing-library/react'

describe('Pruebas en <FirstApp />', () => {
  test('debe coincidir con el snapshot', () => {
    const title = 'Hola, Mundo!'
    const { container } = render(<FirstApp title={title} />)
    expect(container).toMatchSnapshot()
  })

  test('debe mostrar el título en un h1', () => {
    const title = 'Hola, Mundo!'
    const { container, getByText, getByTestId } = render(
      <FirstApp title={title} />
    )

    expect(getByText(title)).toBeTruthy()

    const h1 = container.querySelector('h1')
    expect(h1.innerHTML).toContain(title)

    expect(getByTestId('test-title').innerHTML).toBe(title)
  })

  test('debe mostrar el subtitulo enviado por props', () => {
    const title = 'Hola, Mundo!'
    const subtitle = 'Soy un subtítulo'

    const { getByText } = render(<FirstApp title={title} subtitle={subtitle} />)

    expect(getByText(subtitle)).toBeTruthy()
  })
})
