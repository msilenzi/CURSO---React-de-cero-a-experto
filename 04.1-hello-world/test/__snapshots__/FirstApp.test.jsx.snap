// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe coincidir con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, Mundo!
  </h1>
  <h2>
    No hay subt√≠tulo
  </h2>
  <p>
    Manuel Silenzi
  </p>
</div>
`;
